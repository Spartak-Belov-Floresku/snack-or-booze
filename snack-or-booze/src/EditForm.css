form{
    padding:30px;
    border-radius: 5px;
    background-color: rgba(3, 131, 31, 0.8);
}
input{
    display: block;
    margin:0 auto 1em;
    background-color: chartreuse;
}
label{
    clear: right;
    color: rgb(6, 255, 68);
}