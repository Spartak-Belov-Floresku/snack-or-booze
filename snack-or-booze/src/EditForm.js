import { useState } from "react";
import { Redirect } from "react-router-dom";
import './EditForm.css';
import SnackOrBoozeApi from "./Api";
import useFields from './hooks/useFields';


/**
 * this component helps to update data for snack or drink into data base using HTML from
 */
const EditForm = ({itemRef, item}) => {

    const[ update, setUpdate ] = useState(false)

    const [formData, handleChange, releaseButton ] = useFields({
      id: item.id,
      name: item.name,
      description: item.description,
      recipe: item.recipe,
      serve: item.serve
    });
    

    const handleData = async (data) => {
      await SnackOrBoozeApi.updateData(itemRef, data);
      setUpdate(true);
    }

    const handleSubmit = e => {
        e.preventDefault();
        handleData(formData);
    }

    return update? (<Redirect to={`/${itemRef}/${formData.id}/`} />):(
                                                                      <form onSubmit={handleSubmit}>
                                                                        <label htmlFor="name">Name:</label>
                                                                          <input
                                                                            type="text"
                                                                            name="name"
                                                                            value={formData.name}
                                                                            onChange={handleChange}
                                                                            id="name"
                                                                          />
                                                                      <label htmlFor="description">Description:</label>
                                                                          <input
                                                                            type="text"
                                                                            name="description"
                                                                            value={formData.description}
                                                                            onChange={handleChange}
                                                                            id="description"
                                                                          />
                                                                      <label htmlFor="recipe">Recipe:</label>
                                                                          <input
                                                                            type="text"
                                                                            name="recipe"
                                                                            value={formData.recipe}
                                                                            onChange={handleChange}
                                                                            id="recipe"
                                                                          />
                                                                      <label htmlFor="serve">Serve:</label>
                                                                          <input
                                                                            type="text"
                                                                            name="serve"
                                                                            value={formData.serve}
                                                                            onChange={handleChange}
                                                                            id="serve"
                                                                          />
                                                                        {!releaseButton()? <button disabled>Edit {itemRef}</button>: <button>Edit {itemRef}</button>}
                                                                      </form>
                                                                    );
}


export default EditForm;