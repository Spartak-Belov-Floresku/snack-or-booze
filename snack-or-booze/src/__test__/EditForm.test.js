import { render, screen, fireEvent} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SnackOrBoozeApi from "../Api";
import EditForm from '../EditForm';

let snack;

/**befor ech tests get data from database */
beforeEach( async () => {
  function getSnack() {
    return SnackOrBoozeApi.getSnack('nachos');
  }   
  
  snack = await getSnack();

});

/** render EditForm component */
test('renders learn react link', () => {

    render(
        <BrowserRouter>
            <EditForm itemRef={'snacks'} item={snack} />
        </BrowserRouter>
    );
  
});

/** Matching snapshot of EditForm components */
it("matches snapshot Home component", () => {  
      
    const {asFragment} = render(
        <BrowserRouter>
            <EditForm itemRef={'snacks'} item={snack} />
        </BrowserRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
});

/** update data into database */
it("check form submission and get update snack", () => {  
      
    const { queryByText, getByLabelText } = render(<BrowserRouter><EditForm itemRef={'snacks'} item={snack} /></BrowserRouter>);

    const btn = queryByText("Edit snacks");
    const input = getByLabelText("Description:");

    //check valu befor change
    expect(input.closest("input").value).toEqual("An American classic!");

    input.closest("input").value = "Update description...!!!!";

    fireEvent.click(btn);

    jest.fn(() => expect(screen.findByText("Update description...!!!")).toBeInTheDocument());
    
});