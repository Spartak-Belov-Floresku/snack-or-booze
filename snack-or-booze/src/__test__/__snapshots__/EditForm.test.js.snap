// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Home component 1`] = `
<DocumentFragment>
  <form>
    <label
      for="name"
    >
      Name:
    </label>
    <input
      id="name"
      name="name"
      type="text"
      value="Nachos"
    />
    <label
      for="description"
    >
      Description:
    </label>
    <input
      id="description"
      name="description"
      type="text"
      value="An American classic!"
    />
    <label
      for="recipe"
    >
      Recipe:
    </label>
    <input
      id="recipe"
      name="recipe"
      type="text"
      value="Cover expensive, organic tortilla chips with Cheez Whiz."
    />
    <label
      for="serve"
    >
      Serve:
    </label>
    <input
      id="serve"
      name="serve"
      type="text"
      value="Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
    />
    <button>
      Edit snacks
    </button>
  </form>
</DocumentFragment>
`;
